packages/dynamite/dynamite/lib/src/builder/client.dart

```diff
@@ -181,308 +182,254 @@ Iterable<Method> buildTags(
-                  dataValue = '_response.body';
-                  dataNeedsAwait = true;
-                } else if (mimeType == 'application/json') {
-                  dataType = result.name;
-                  if (result.name == 'dynamic') {
-                    dataValue = '';
-                  } else if (result.name == 'String') {
-                    dataValue = '_response.body';
-                    dataNeedsAwait = true;
-                  } else if (result is TypeResultEnum || result is TypeResultBase) {
-                    dataValue = result.deserialize(result.decode('await _response.body'));
-                    dataNeedsAwait = false;
-                  } else {
-                    dataValue = result.deserialize('await _response.jsonBody');
-                    dataNeedsAwait = false;
-                  }
-                } else {
-                  throw Exception('Can not parse mime type "$mimeType"');
-                }
-              }
-            }
+        code.writeln('''
+  return DynamiteRawResponse<$returnDataType, $returnHeadersType>(
+    response: $client.doRequest(
+      '$httpMethod',
+      _uri,
+      _headers,
+      _body,
+      validateStatus: validateStatus,
+    ),
+    bodyType: ${dataType?.fullType},
+    headersType: ${headersType?.fullType},
+    serializers: _jsonSerializers,
+  );
+''');
+      }
 
-            if (headersType != null && dataType != null) {
-              b.returns = refer('Future<DynamiteResponse<$dataType, $headersType>>');
-              code.write(
-                'return DynamiteResponse<$dataType, $headersType>(${dataNeedsAwait ?? false ? 'await ' : ''}$dataValue, $headersValue,);',
-              );
-            } else if (headersType != null) {
-              b.returns = refer('Future<$headersType>');
-              code.write('return $headersValue;');
-            } else if (dataType != null) {
-              b.returns = refer('Future<$dataType>');
-              code.write('return $dataValue;');
-            } else {
-              b.returns = refer('Future<void>');
-              code.write('return;');
-            }
+      yield Method((final b) {
+        b
+          ..name = name
+          ..modifier = MethodModifier.async
+          ..docs.addAll(docs);
+
+        if (annotations != null) {
+          b.annotations.add(annotations);
+        }
+
+        final parameters = operationParameters.build();
+        final rawParameters = parameters.map((final p) => '${p.name}: ${p.name},').join('\n');
+
+        b
+          ..optionalParameters.addAll(parameters)
+          ..returns = refer('Future<DynamiteResponse<$returnDataType, $returnHeadersType>>')
+          ..body = Code('''
+final rawResponse = ${name}Raw(
+  $rawParameters
+);
+
+await rawResponse.future;
+
+return rawResponse.response;
+''');
+      });
+
+      yield Method(
+        (final b) {
+          b
+            ..name = '${name}Raw'
+            ..docs.addAll(docs);
```

Comment:
Please add the comment that this method should only be used if you are know what you are doing and that the other one should probably be used instead.